src/views/SystemStatus.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useState } from 'react'
+import axios from 'axios'
+import Table from 'react-bootstrap/Table'
+import { useCookies } from 'react-cookie'
+import PropTypes from 'prop-types'
+import checkIcon from '../graphics/check-circle-fill.svg'
+import xIcon from '../graphics/x-circle-fill.svg'
+import { Environments } from '../utilities/Constants'
+
+export default function SystemStatus(props) {
+  const { environment } = props
+  const [sensorServerConnected, setSensorServerConnected] = useState(false)
+  const [buttonServerConnected, setButtonServerConnected] = useState(false)
+  const [sensorDatabaseConnected, setSensorDatabaseConnected] = useState(false)
+  const [buttonsDatabaseConnected, setButtonsDatabaseConnected] = useState(false)
+  const [cookies] = useCookies(['googleIdToken'])
+  let braveApiKey = ''
+  let sensorBaseUrl = ''
+  let buttonsBaseUrl = ''
+
+  if (environment === Environments.dev.name) {
+    sensorBaseUrl = process.env.REACT_APP_SENSOR_DEV_URL
+    buttonsBaseUrl = process.env.REACT_APP_BUTTONS_DEV_URL
+    braveApiKey = process.env.REACT_APP_BRAVE_API_KEY_DEV
+  } else if (environment === Environments.prod.name) {
+    sensorBaseUrl = process.env.REACT_APP_SENSOR_PROD_URL
+    buttonsBaseUrl = process.env.REACT_APP_BUTTONS_PROD_URL
+    braveApiKey = process.env.REACT_APP_BRAVE_API_KEY_PROD
+  } else if (environment === Environments.staging.name) {
+    sensorBaseUrl = process.env.REACT_APP_SENSOR_STAGING_URL
+    buttonsBaseUrl = process.env.REACT_APP_BUTTONS_STAGING_URL
+    braveApiKey = process.env.REACT_APP_BRAVE_API_KEY_STAGING
+  }
+
+  async function getSensorServerStatus() {
+    try {
+      await axios({
+        method: 'GET',
+        url: `${sensorBaseUrl}/login`,
+        headers: {
+          'Cache-Control': 'no-store', // Disable caching so request has the most up-to-date status
+        },
+      })
+      setSensorServerConnected(true)
+    } catch (error) {
```

Comment:
If the server responds with 404, or another error code, does axios throw an error? Maybe I'm remember incorrectly, but I thought it would simply return the server response, which can be checked with:
```javascript
const res = await axios.get(...);
if (res.status === 200) {
  // ok response
} else {
  // not ok response
}
```
Anyways, I just want to double check that any HTTP response that isn't 200 is considered a disconnected server response.

I'll extend this question to the other checks.

Reply:
Yes axios will throw an error for a non-200 status. I've made sure to test this as well by changing the URL to my localhost, and then running (and then not running) BraveSensor and BraveButtons locally. When the GET request 404's, the code goes to the catch block, logs the error and visually shows a red x.

Reply:
Nice! Good to know.